### PR TITLE
Fix Dns tests on OSX.

### LINKED ISSUE #23824: Test: System.Net.NameResolution.Tests.GetHostByNameTest/DnsObsoleteBeginEndGetHostByName_EmptyString_ReturnsHostName failed with "Xunit.Sdk.ContainsException"
Opened on behalf of @Jiayili1

The test `System.Net.NameResolution.Tests.GetHostByNameTest/DnsObsoleteBeginEndGetHostByName_EmptyString_ReturnsHostName` has failed.

Assert.Contains() Failure
    Not found: DCI-Mac-Build-068.local
    In value:  dci-mac-build-068.local

        Stack Trace:

           at System.Net.NameResolution.Tests.GetHostByNameTest.DnsObsoleteBeginEndGetHostByName_EmptyString_ReturnsHostName() in /Users/buildagent/agent/_work/30/s/corefx/src/System.Net.NameResolution/tests/FunctionalTests/GetHostByNameTest.cs:line 115
Build : Master - 20170906.01 (Core Tests)
Failing configurations:
- OSX.1012.Amd64-x64
  - Release

Detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170906.01/workItem/System.Net.NameResolution.Functional.Tests/analysis/xunit/System.Net.NameResolution.Tests.GetHostByNameTest~2FDnsObsoleteBeginEndGetHostByName_EmptyString_ReturnsHostName

### PR DESCRIPTION
**Pull request by Priya91**
fixes #23824 

cc @danmosemsft

### CODE REVIEW

File: src/System.Net.NameResolution/tests/FunctionalTests/GetHostByNameTest.cs [hunk 1]
```diff
@@ -106,14 +106,14 @@ public void DnsObsoleteGetHostByName_IPv6String_ReturnsOnlyGivenIP()
         public void DnsObsoleteGetHostByName_EmptyString_ReturnsHostName()
         {
             IPHostEntry entry = Dns.GetHostByName("");
-            Assert.Contains(Dns.GetHostName(), entry.HostName);
+            Assert.Contains(Dns.GetHostName(), entry.HostName, StringComparison.OrdinalIgnoreCase);
```
Comment (danmosemsft) [inline src/System.Net.NameResolution/tests/FunctionalTests/GetHostByNameTest.cs]: It should be case insensitive for just a-z, based on that RFC. @tarekgh is there a built in way to be ASCII case insensitive, sensitive otherwise? If not you can trivially do it (in almost one line) of course.

Comment (danmosemsft) [inline src/System.Net.NameResolution/tests/FunctionalTests/GetHostByNameTest.cs]: can you add a link to the RFC maybe

Comment (danmosemsft) [reply src/System.Net.NameResolution/tests/FunctionalTests/GetHostByNameTest.cs]: Looks like there isn't a built in way, on ASCIIEncoding.

File: src/System.Net.NameResolution/tests/FunctionalTests/GetHostByNameTest.cs [hunk 2]
```diff
@@ -106,13 +106,17 @@ public void DnsObsoleteGetHostByName_IPv6String_ReturnsOnlyGivenIP()
         public void DnsObsoleteGetHostByName_EmptyString_ReturnsHostName()
         {
             IPHostEntry entry = Dns.GetHostByName("");
+
+            // Dns ASCII comparisons are case insensitive. Refer RFC4343
```
Comment (davidsh) [inline src/System.Net.NameResolution/tests/FunctionalTests/GetHostByNameTest.cs]: I would rephrase to this:

"DNS labels should be compared as case insensitive for ASCII characters.  See RFC 4343."

**Comment by Priya91** [comment]
Linux build failed with linq and drawing test failures.
